File: The Brave Hero.cpp
#include <bits/stdc++.h>
using namespace std;
struct Carl{short viata,putere,aparare,viteza,noroc,abilitate1,abilitate2;};
struct Bestie{short viata,putere,aparare,viteza,noroc;};
Carl carl;
Bestie bestie;
bool ok;
short dodge,NumarTure;
int main()
{
    srand(time(0));

    carl.viata = rand() % 31 + 65; cout<<"Viata lui Carl este: "<<carl.viata<<'\n';
    carl.putere = rand() % 11 + 60; cout<<"Puterea lui Carl este: "<<carl.putere<<'\n';
    carl.aparare = rand() % 11 + 40; cout<<"Apararea lui Carl este: "<<carl.aparare<<'\n';
    carl.viteza = rand() % 11 + 40; cout<<"Viteza lui Carl este: "<<carl.viteza<<'\n';
    carl.noroc = rand() % 21 + 10; cout<<"Norocul lui Carl este: "<<carl.noroc<<'\n';

    cout<<'\n'<<'\n'<<'\n';

    bestie.viata = rand() % 26 + 55; cout<<"Viata bestiei este: "<<bestie.viata<<'\n';
    bestie.putere = rand() % 31 + 50; cout<<"Puterea bestiei este: "<<bestie.putere<<'\n';
    bestie.aparare = rand() % 21 + 35; cout<<"Apararea bestiei este: "<<bestie.aparare<<'\n';
    bestie.viteza = rand() % 21 + 40; cout<<"Viteza bestiei este: "<<bestie.viteza<<'\n';
    bestie.noroc = rand() % 16 + 25; cout<<"Norocul bestiei este: "<<bestie.noroc<<'\n';

    cout<<'\n'<<'\n'<<'\n';

    if(carl.viteza > bestie.viteza) { ok = 1; cout<<"Carl are primul atac."<<'\n'<<'\n'; } /// Carl are primul atac
    else if(bestie.viteza > carl.viteza) { ok = 0; cout<<"Bestia are primul atac."<<'\n'<<'\n'; }/// Bestia incepe
    else{
         if(carl.noroc >= bestie.noroc) { ok = 1; cout<<"Carl are primul atac."<<'\n'<<'\n'; } /// O sa presupun ca incepe Carl si daca au norocul egal
         else { ok = 0 ; cout<<"Bestia are primul atac."<<'\n'<<'\n'; }
    }
    while(carl.viata > 0 && bestie.viata > 0 && NumarTure < 20)
    {
        NumarTure++;
        cout<<"Inceputul turului cu numarul "<<NumarTure<<":"<<'\n';
        if(ok == 1)
        {
            cout<<"    Este randul lui Carl sa atace."<<'\n';
            dodge = rand() % 100 + 1;
            if(dodge <= bestie.noroc) cout<<"    Bestia are noroc si se fereste de atac."<<'\n'; /// Bestie a avut noroc sa dea dodge
            else{
                carl.abilitate1 = rand() % 100 + 1;
                if(carl.abilitate1 <= 10) /// Atacul lui Carl se dubleaza pentru urmatorul atac
                    {
                     cout<<"    Carl foloseste forta dragonului si isi dubleaza puterea pentru urmatorul atac."<<'\n';
                     if(carl.putere * 2 - bestie.aparare <= 0) cout<<"    Atacul lui Carl nu ii provoaca daune bestiei."<<'\n';
                        else if(carl.putere * 2 - bestie.aparare >= 100){ bestie.viata -= 100; cout<<"    Atacul lui Carl ii provoaca bestiei daune in valoare de 100."<<'\n'; }
                            else { bestie.viata = bestie.viata - (carl.putere * 2 - bestie.aparare); cout<<"    Atacul lui Carl ii provoaca bestiei daune in valoare de "<<carl.putere * 2 - bestie.aparare<<"."<<'\n'; }
                    }
                else{
                    if(carl.putere - bestie.aparare <= 0) cout<<"    Atacul lui Carl nu ii provoaca daune bestiei."<<'\n';
                        else if(carl.putere - bestie.aparare >= 100){ bestie.viata -= 100; cout<<"    Atacul lui Carl ii provoaca bestiei daune in valoare de 100."<<'\n'; }
                            else { bestie.viata = bestie.viata - (carl.putere - bestie.aparare); cout<<"    Atacul lui Carl ii provoaca bestiei daune in valoare de "<<carl.putere - bestie.aparare<<"."<<'\n'; }
                }
            }
            cout<<"    Bestia ramane cu "<<bestie.viata<<" puncte de viata la finalul turului."<<'\n';
            ok = 0; /// Urmatorul atac o sa fie al bestiei
        }
        else{
            cout<<"    Este randul bestiei sa atace."<<'\n';
            dodge = rand() % 100 + 1;
            if(dodge <= carl.noroc)  cout<<"    Carl are noroc si se fereste de atac."<<'\n';
            else{
                carl.abilitate2 = rand() % 100 + 1;
                if(carl.abilitate2 <= 20) /// Atacul bestiei se injumatateste pentru urmatorul atac
                    {
                     cout<<"    Carl foloseste scutul fermecat si injumatateste daunele urmatorului atac."<<'\n';
                     if(bestie.putere / 2 - carl.aparare <= 0) cout<<"    Atacul bestiei nu ii provoaca daune lui Carl."<<'\n';
                        else if(bestie.putere / 2 - carl.aparare >= 100){ carl.viata -= 100; cout<<"    Atacul bestiei ii provoaca lui Carl daune in valoare de 100."<<'\n'; }
                            else { carl.viata = carl.viata - (bestie.putere / 2 - carl.aparare); cout<<"    Atacul bestiei ii provoaca lui Carl daune in valoare de "<<bestie.putere / 2 - carl.aparare<<"."<<'\n'; }
                    }
                else{
                    if(bestie.putere - carl.aparare <= 0) cout<<"    Atacul bestiei nu ii provoaca daune lui Carl."<<'\n';
                        else if(bestie.putere - carl.aparare >= 100){ carl.viata -= 100; cout<<"    Atacul bestiei ii provoaca lui Carl daune in valoare de 100."<<'\n'; }
                            else { carl.viata = carl.viata - (bestie.putere - carl.aparare); cout<<"    Atacul bestiei ii provoaca lui Carl daune in valoare de "<<bestie.putere - carl.aparare<<"."<<'\n'; }
                }
            }
            cout<<"    Carl ramane cu "<<carl.viata<<" puncte de viata la finalul turului."<<'\n';
            ok = 1; /// Urmatorul atac o sa fie al lui Carl

        }
        cout<<'\n';
    }
    if(carl.viata <= 0) cout<<"Bestia l-a infrant pe Carl dupa "<<NumarTure<<" ture, ramanand cu "<<bestie.viata<<" puncte de viata."<<'\n';
        else if(bestie.viata <= 0) cout<<"Carl a infrant bestia dupa "<<NumarTure<<" ture, ramanand cu "<<carl.viata<<" puncte de viata."<<'\n';
            else if(NumarTure == 20) cout<<"Cele 20 de ture s-au terminat si Carl a ramas cu "<<carl.viata<<" puncte de viata si bestia cu "<<bestie.viata<<" puncte de viata."<<'\n';
}

File: TheBraveHero.py
import random

carl = {"Viata": random.randint(65, 95), "Putere": random.randint(60, 70), "Aparare": random.randint(40, 50), "Viteza": random.randint(40, 50), "Noroc": random.randint(10, 30)}

print("Viata lui Carl este:", carl["Viata"])
print("Puterea lui Carl este:", carl["Putere"])
print("Apararea lui Carl este:", carl["Aparare"])
print("Viteza lui Carl este:", carl["Viteza"])
print("Norocul lui Carl este:", carl["Noroc"])

print()

bestie = {"Viata": random.randint(55, 90), "Putere": random.randint(50, 80), "Aparare": random.randint(35, 55), "Viteza": random.randint(40, 60), "Noroc": random.randint(25, 40)}

print("Viata bestiei este:", bestie["Viata"])
print("Puterea bestiei este:", bestie["Putere"])
print("Apararea bestiei este:", bestie["Aparare"])
print("Viteza bestiei este:", bestie["Viteza"])
print("Norocul bestiei este:", bestie["Noroc"])

print()

if carl["Viteza"] > bestie["Viteza"]:
    ok = 1
    print("Carl ataca primul.", '\n')
elif bestie["Viteza"] > carl["Viteza"]:
    ok = 0
    print("Bestia ataca prima.", '\n')
else:
    if carl["Noroc"] >= bestie["Noroc"]:
        ok = 1
        print("Carl ataca primul.", '\n')
    else:
        ok = 0
        print("Bestia ataca prima.", '\n')

NumarTure = 0

while carl["Viata"] > 0 and bestie["Viata"] > 0 and NumarTure < 20:
    NumarTure += 1
    print("Inceputul turului cu numarul {0}:".format(NumarTure))
    if ok == 1:
        print("\tEste randul lui Carl sa atace.")

        dodge = random.randint(1, 100)

        if dodge <= bestie["Noroc"]:
            print("\tBestia are noroc si se fereste de atac.")

        else:
            abilitate = random.randint(1, 100)

            if abilitate <= 10:
                print("\tCarl foloseste forta dragonului pentru a-si dubla puterea.")

                if carl["Putere"] * 2 - bestie["Aparare"] <= 0:
                    print("\tAtacul lui Carl nu ii provoaca daune bestiei.")

                elif carl["Putere"] * 2 - bestie["Aparare"] >= 100:
                    bestie["Viata"] -= 100
                    print("\tAtacul lui Carl ii provoaca bestiei daune in valoare de 100.")

                else:
                    bestie["Viata"] = bestie["Viata"] - (carl["Putere"] * 2 - bestie["Aparare"])
                    print("\tAtacul lui Carl ii provoaca bestiei daune in valoare de {0}.".format(carl["Putere"] * 2 - bestie["Aparare"]))

            else:
                if carl["Putere"] - bestie["Aparare"] <= 0:
                    print("\tAtacul lui Carl nu ii provoaca daune bestiei.")

                elif carl["Putere"] - bestie["Aparare"] >= 100:
                    bestie["Viata"] -= 100
                    print("\tAtacul lui Carl ii provoaca bestiei daune in valoare de 100.")

                else:
                    bestie["Viata"] = bestie["Viata"] - (carl["Putere"] - bestie["Aparare"])
                    print("\tAtacul lui Carl ii provoaca bestiei daune in valoare de {0}.".format(carl["Putere"] - bestie["Aparare"]))

        ok = 0
        print("\tBestia ramane cu {0} puncte de viata.".format(bestie["Viata"]))

    else:
        print("\tEste randul bestiei sa atace.")

        dodge = random.randint(1, 100)

        if dodge <= carl["Noroc"]:
            print("\tCarl are noroc si se fereste de atac.")

        else:
            abilitate = random.randint(1, 100)

            if abilitate <= 20:
                print("\tCarl foloseste scutul fermecat pentru a-si injumatati daunele primite.")

                if bestie["Putere"] // 2 - carl["Aparare"] <= 0:
                    print("\tAtacul bestiei nu ii provoaca daune lui Carl.")

                elif bestie["Putere"] // 2 - carl["Aparare"] >= 100:
                    carl["Viata"] -= 100
                    print("\tAtacul bestiei ii provoaca daune lui Carl in valoare de 100.")

                else:
                    carl["Viata"] = carl["Viata"] - (bestie["Putere"] // 2 - carl["Aparare"])
                    print("\tAtacul bestiei ii provoaca lui Carl daune in valoare de {0}.".format(bestie["Putere"] // 2 - carl["Aparare"]))

            else:
                if bestie["Putere"] - carl["Aparare"] <= 0:
                    print("\tAtacul bestiei nu ii provoaca daune lui Carl.")

                elif bestie["Putere"] - carl["Aparare"] >= 100:
                    carl["Viata"] -= 100
                    print("\tAtacul bestiei ii provoaca daune lui Carl in valoare de 100.")

                else:
                    carl["Viata"] = carl["Viata"] - (bestie["Putere"] - carl["Aparare"])
                    print("\tAtacul bestiei ii provoaca lui Carl daune in valoare de {0}.".format(bestie["Putere"] - carl["Aparare"]))

        ok = 1
        print("\tCarl ramane cu {0} puncte de viata.".format(carl["Viata"]))

    print()

if carl["Viata"] <= 0:
    print("Bestia l-a infrant pe Carl dupa {0} ture, ramanand cu {1} puncte de viata.".format(NumarTure, bestie["Viata"]))

elif bestie["Viata"] <= 0:
    print("Carl a infrant bestia dupa {0} ture, ramanand cu {1} puncte de viata.".format(NumarTure, carl["Viata"]))

elif NumarTure == 20:
    print("Cele 20 de ture s-au terminat si Carl a ramas cu {0} puncte de viata si bestia cu {1} puncte de viata.".format(carl["Viata"], bestie["Viata"]))

